src/imageLoader/internal/xhrRequest.js

```diff
@@ -16,8 +16,46 @@ function xhrRequest (url, imageId, headers = {}) {
     xhr.setRequestHeader(key, headers[key]);
   });
 
+  params.deferred = deferred;
+  params.url = url;
+  params.imageId = imageId;
+
+  // Event triggered when downloading an image starts
+  xhr.onloadstart = function (event) {
+    // Action
+    if (options.onloadstart) {
+      options.onloadstart(event, params);
+    }
+
+    // Event
+    $(cornerstone.events).trigger('CornerstoneImageLoadStart', {
+      url
+    });
+  };
+
+  // Event triggered when downloading an image ends
+  xhr.onloadend = function (event) {
+    // Action
+    if (options.onloadend) {
+      options.onloadend(event, params);
+    }
+
+    // Event
+    $(cornerstone.events).trigger('CornerstoneImageLoadEnd', {
```

Comment:
Can you add the imageId to the object for CornerstoneImageLoadStart and CornerstoneImageLoadEnd? Otherwise this looks good to merge. Thanks!

Reply:
Done :)